src/providers/wfs/qgswfsprovider.cpp

```diff
@@ -1258,7 +1258,7 @@ bool QgsWFSProvider::empty() const
 {
   QgsFeature f;
   QgsFeatureRequest request;
-  request.setSubsetOfAttributes( QgsAttributeList() );
+  request.setNoAttributes();
   request.setFlags( QgsFeatureRequest::NoGeometry );
```

Comment:
@3nids This is risky (at least using PyQGIS) because the line:

`request.setFlags( QgsFeatureRequest::NoGeometry );`

seems to reset the call to:

`request.setNoAttributes(); `

For example, on a layer with a single string field, using this code snippet:

```
layer = iface.activeLayer()
r = QgsFeatureRequest()
r.setNoAttributes()
r.setFlags(QgsFeatureRequest.NoGeometry)
res = [f for f in layer.getFeatures(r)]
print(res[0].attributes(), "g:", res[0].geometry().asWkt())
```

I get:

`['abc'] g:`

But inverting the order of  `r.setNoAttributes()` and `r.setFlags(QgsFeatureRequest.NoGeometry)` lines, I get:

`[None] g:`

As you can see, the geometry is never retrieved, but depending on the order of those 2 lines, attributes may come even if we don't want them. 

I'm afraid that this may also be happenning in C++, and so, QGIS would always be getting all attributes when placing `setNoAttributes()` before setting the flag `NoGeometry`.